### ISSUE TITLE
strawberry-graphql package dependency cleanup

### ISSUE THREAD

**Issue by la4de** [strawberry-graphql/strawberry#1432]
I had a fun week with pyodide. I implemented simple playground demo few days ago and proper version later during the weekend. It really was fun! Strawberry GraphQL Playground is available here https://la4de.github.io/strawberry-playground/.

During that work I noticed that `strawberry-graphql` package has many dependencies which are not actually used by the core. I removed most of them manually for playground to speed up the loading. The most problematic dependency is `python-multipart` which does not seem to have pure python implementation. However, that's not used by playground so I removed that too.

After rough clean up I ended removing following dependencies.
```diff
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -27,24 +27,11 @@ build-backend = "poetry.masonry.api"

 [tool.poetry.dependencies]
 python = "^3.7"
-starlette = {version = ">=0.13.6,<0.17.0", optional = true}
-click = {version = ">=7.0,<9.0", optional = true}
 pygments = "^2.3"
-uvicorn = {version = ">=0.11.6,<0.16.0", optional = true}
-django = {version = ">=2,<4", optional = true}
 graphql-core = "~3.1.0"
-asgiref = {version = "^3.2", optional = true}
-flask = {version = "^1.1", optional = true}
 typing_extensions = "^3.7.4"
-opentelemetry-api = {version = "<2", optional = true}
-opentelemetry-sdk = {version = "<2", optional = true}
 python-dateutil = "^2.7.0"
 cached-property = "^1.5.2"
-pydantic = {version = "<2", optional = true}
-python-multipart = "^0.0.5"
-sanic = {version = ">=20.12.2,<22.0.0", optional = true}
-aiohttp = {version = "^3.7.4.post0", optional = true}
-fastapi = {version = ">=0.65.2", optional = true}
 sentinel = "^0.3.0"

 [tool.poetry.dev-dependencies]
 ```

It would be nice to keep mandatory dependencies at minimum level.

**Comment by patrick91** [comment]
all the ones that are removed seems to be optional, with the exception of python-multipart

does pyodide try to install the optional ones too? 🤔

**Comment by la4de** [comment]
> all the ones that are removed seems to be optional, with the exception of python-multipart
> 
> does pyodide try to install the optional ones too? 🤔

You are right. Optional packages are not installed by `micropip` from `pyodide`. I went through only the package names and did not read versions or other attributes.

`python-multipart` is actually pure python package. Installation of `python-multipart` still fails because there is only source dist version in pypi. `micropip` can install only pure python 3 wheel packages.

There is a ticket under that project which requests wheel package. I added my comment there too :)
https://github.com/andrew-d/python-multipart/issues/37
